Entity-ngsiv2-schema.json

```diff
@@ -0,0 +1,185 @@
+{
+  "$schema": "http://json-schema.org/schema#",
+  "id": "https://fiware.github.io/dataModels/Entity-ngsiv2-schema.json",
+  "title": "NGSIv2 Entity",
+  "description": "NGSIv2 Entity",
+  "definitions": {
+    "EntityIdentifier": {
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 256,
+      "pattern": "^[\\w\\-\\.\\{\\}\\$\\+\\*\\[\\]`|~^@!,:\\\\]+$"
+    },
+    "Name": {
+      "type": "string",
+      "pattern": "^(\\d|[a-zA-Z]|_)+$",
+      "minLength": 1,
+      "description": "NGSIv2 Name"
+    },
+    "Value": {
+      "type": [
+        "number",
+        "boolean",
+        "array",
+        "object",
+        "null",
+        "string"
+      ]
+    },
+    "TypeValue": {
```

Comment:
I think we should distinguish between *entity*, *attribute*, *metadata* types given they have different constrains. In particular, the special attributes for dates and geo-location only applies to *attributes*, as far as I remember.

In addition, the following sections in the [NGSIv2 specification](http://telefonicaid.github.io/fiware-orion/api/v2/stable/) should be taking into account:

"Field syntax restrictions"

Reply:
see https://github.com/Fiware/dataModels/pull/373/files#diff-c4a04684aa22645d0665ae4a1afc7cf0R88
and
https://github.com/Fiware/dataModels/pull/373/files#diff-c4a04684aa22645d0665ae4a1afc7cf0R107

Reply:
Thanks for the links!

Ok, I think I understand how it works. With `oneOf` you combine the basic type with the special types in each case (DateTimeTypeValue in the case of metadata and DateTimeTypeValue  and GeoTypeValue in the case of attributes).

Thus NTC with regards this comment (although the incoroporation of additional geo: types is pending in [another comment](https://github.com/Fiware/dataModels/pull/373#discussion_r255896352)).

(Btw looking into this I have realized there is a small documentation flaw in NGSIv2 spec. See https://github.com/telefonicaid/fiware-orion/issues/3434. To be fixed)

Reply:
NTC